orchagent/routeorch.cpp

```diff
@@ -720,22 +731,32 @@ void RouteOrch::doTask(Consumer& consumer)
                             it++;
                     }
                 }
+                /*
+                 * Check if the route does not exist or needs to be updated or
+                 * if the route is using a temporary next hop group owned by
+                 * NhgOrch.
+                 */
                 else if (m_syncdRoutes.find(vrf_id) == m_syncdRoutes.end() ||
                     m_syncdRoutes.at(vrf_id).find(ip_prefix) == m_syncdRoutes.at(vrf_id).end() ||
-                    m_syncdRoutes.at(vrf_id).at(ip_prefix) != nhg)
+                    m_syncdRoutes.at(vrf_id).at(ip_prefix) != RouteNhg(nhg, ctx.nhg_index) ||
+                    ctx.using_temp_nhg)
                 {
                     if (addRoute(ctx, nhg))
                         it = consumer.m_toSync.erase(it);
                     else
                         it++;
                 }
                 else
+                {
+                    SWSS_LOG_INFO("Route %s is duplicate entry", key.c_str());
```

Comment:
@prsunny Will this hog the syslog. Is this useful log?

Reply:
INFO is disabled by default. So should be fine. But i dont find it as a useful log. Please remove

Reply:
Done.